src/wui/story_message_box.cc

```diff
@@ -62,10 +80,20 @@ StoryMessageBox::StoryMessageBox(Widelands::Game* game,
 
 	main_box_.add(&textarea_, UI::Box::Resizing::kExpandBoth);
 	main_box_.add(&button_box_, UI::Box::Resizing::kFullSize);
+
 	button_box_.add_inf_space();
 	button_box_.add(&ok_);
 	button_box_.add_inf_space();
 
+	if (allow_next_scenario) {
+		button_box_.add(&next_scenario_);
+		button_box_.add_inf_space();
+		next_scenario_.sigclicked.connect([this]() { clicked_next_scenario(); });
+		ok_.set_tooltip(_("Continue playing this mission"));
+	} else {
+		next_scenario_.set_visible(false);
+	}
```

Comment:
Can we have the "Main menu" button here? Then it is always possible to directly quit the scenario, either to the next one, or to the main menu.
The ok tooltip should be outside the condition.